### PR TITLE
Challenge26 merge sort

### PR DESCRIPTION
**Pull request by Kozer2**

### CODE REVIEW

**Comment by Kozer2** [comment]
- [ ] Top-level README “Table of Contents” is updated
 - [ ] Feature tasks for this challenge are completed
 - [ ] Unit tests written and passing
     - [ ] “Happy Path” - Expected outcome
     - [ ] Expected failure
     - [ ] Edge Case (if applicable/obvious)
 - [X] README for this challenge is complete
     - [X] Summary, Description, Approach & Efficiency, Solution
     - [X] Link to code
     - [X] Picture of whiteboard

File: CHallenges.Tests/ArrayChallenges/MergeSortTests.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            // arrange
+            int[] arr = new int[6] { 8, 4, 23, 42, 16, 15 };
+
+
+            // act
+            int[] result = MergeSortChallenge.MergeSort(arr);
+
+
+            //assert
+            Assert.Equal(arr, result);
```
Comment (dahlbyk) [inline CHallenges.Tests/ArrayChallenges/MergeSortTests.cs:25]: Need a better `expected` here of input actually sorted!

File: Challenges/ArrayChallenges/MergeSortChallenge.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        public static int[] MergeSort(int[] arr)
+        {
+            int n = arr.Length;
+
+            
+            if(n > 1)
+            {
+                int mid = arr.Length / 2;
+                int[] left = new int[mid];
+                int[] right = new int[mid];
```
Comment (dahlbyk) [inline Challenges/ArrayChallenges/MergeSortChallenge.cs:20]: ⚠️ If array length is odd, you want `n - mid`

File: Challenges/ArrayChallenges/MergeSortChallenge.cs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+   public class MergeSortChallenge
+    {
+        public static int[] MergeSort(int[] arr)
+        {
+            int n = arr.Length;
+
+            
+            if(n > 1)
+            {
+                int mid = arr.Length / 2;
```
Comment (dahlbyk) [inline Challenges/ArrayChallenges/MergeSortChallenge.cs:18]: Can use `n` here instead of `arr.Length` again.

File: Challenges/ArrayChallenges/MergeSortChallenge.cs [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+                int[] left = new int[mid];
+                int[] right = new int[mid];
+                for (int i = 0; i < mid; i++)
+                {
+                    left[i] = arr[i];
+                }
+                int x = 0;
+                for (int i = mid; i < arr.Length; i++)
```
Comment (dahlbyk) [inline Challenges/ArrayChallenges/MergeSortChallenge.cs:26]: Length = 7
mid = 3
left.Length = 3 (indexes 0..2)
right.Length = 4 (indexes 3..6)

File: Challenges/ArrayChallenges/MergeSortChallenge.cs [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+            int i = 0;
+            int j = 0;
+            int k = 0;
+
+            while (i < left.Length && j < right.Length)
+               {
+                if (left[i] <= right[j])
+                {
+                    arr[k] = right[j];
+                    j =+ 1;
```
Comment (dahlbyk) [inline Challenges/ArrayChallenges/MergeSortChallenge.cs:51]: `j += 1`

File: Challenges/ArrayChallenges/MergeSortChallenge.cs [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        {
+            int i = 0;
+            int j = 0;
+            int k = 0;
+
+            while (i < left.Length && j < right.Length)
+               {
+                if (left[i] <= right[j])
+                {
+                    arr[k] = right[j];
```
Comment (dahlbyk) [inline Challenges/ArrayChallenges/MergeSortChallenge.cs:50]: If `left[i]` is smaller, we want `arr[k] = left[i]`.

File: Challenges/ArrayChallenges/MergeSortChallenge.cs [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+                    j =+ 1;
+                }
+                k += 1;
```
Comment (dahlbyk) [inline Challenges/ArrayChallenges/MergeSortChallenge.cs:53]: Missing an `else` here to copy `right[j]` into `arr[k]`.

**Review by dahlbyk — COMMENTED** [review]
Rambling nonsense
